server/src/com/google/refine/Refine.java

```diff
@@ -494,12 +494,42 @@ public void actionPerformed(ActionEvent e) {
     
     private void openBrowser() {
         if (!Desktop.isDesktopSupported()) {
-            logger.warn("Java Desktop class not supported on this platform.  Please open %s in your browser",uri.toString());
+            try {
+                openBrowser2();
+            } catch (IOException e) {
+                logger.error("Failed to open browser. Please open %s in your browser",uri.toString());
```

Comment:
Looking at your code, `openBrowser2` can fail without throwing an `IOException` at the moment, so this message might not be displayed in those cases. The control flow of the two methods could probably be adapted to avoid this.